src/test/java/org/opensearch/flowframework/rest/FlowFrameworkResourceSharingRestApiIT.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.rest;
+
+import org.apache.hc.core5.http.HttpHost;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.Response;
+import org.opensearch.client.ResponseException;
+import org.opensearch.client.RestClient;
+import org.opensearch.commons.rest.SecureRestClientBuilder;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.FlowFrameworkRestTestCase;
+import org.opensearch.flowframework.TestHelpers;
+import org.opensearch.flowframework.model.Template;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_ID;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_RESOURCE_TYPE;
+import static org.opensearch.flowframework.common.CommonValue.WORKFLOW_STATE_RESOURCE_TYPE;
+
+public class FlowFrameworkResourceSharingRestApiIT extends FlowFrameworkRestTestCase {
+
+    String aliceUser = "alice";
+    RestClient aliceClient;
+    String bobUser = "bob";
+    RestClient bobClient;
+    String catUser = "cat";
+    RestClient catClient;
+    String dogUser = "dog";
+    RestClient dogClient;
+    String elkUser = "elk";
+    RestClient elkClient;
+    String fishUser = "fish";
+    RestClient fishClient;
+    String lionUser = "lion";
+    RestClient lionClient;
+    private String indexAllAccessRole = "index_all_access";
+    private static String FLOW_FRAMEWORK_FULL_ACCESS_ROLE = "flow_framework_full_access";
+    private static String ML_COMMONS_FULL_ACCESS_ROLE = "ml_full_access";
+    private static String FLOW_FRAMEWORK_READ_ACCESS_ROLE = "flow_framework_read_access";
+
+    // Sharing action-group names for Flow Framework (mirror AD style)
+    private static final String WORKFLOW_READ_ONLY_AG = "workflow_read_only";
+    private static final String WORKFLOW_FULL_ACCESS_AG = "workflow_full_access";
+    private static final String WORKFLOW_STATE_READ_ONLY_AG = "workflow_state_read_only";
+    private static final String WORKFLOW_STATE_FULL_ACCESS_AG = "workflow_state_full_access";
+
+    @Before
+    public void setupSecureTests() throws IOException {
+        if (!isHttps()) throw new IllegalArgumentException("Secure Tests are running but HTTPS is not set");
+
+        // If the suite is launched without the flag, just skip these tests cleanly.
+        assumeTrue("RS tests only run when resource_sharing.enabled=true", isResourceSharingFeatureEnabled());
```

Comment:
Probably need to reverse the order of these statements so we don't throw IAE when resource sharing isn't enabled, which is currently failing all integ tests.